core/src/test/java/org/apache/brooklyn/core/entity/EntityConfigTest.java

```diff
@@ -208,8 +210,18 @@ public void testGetConfigMapWithExplicitMap() throws Exception {
         assertEquals(entity.config().getNonBlocking(TestEntity.CONF_MAP_THING.subKey("mysub")).get(), "myval");
     }
     
-    @Test
-    public void testGetConfigNonBlocking() throws Exception {
+    // TODO This now fails because the task has been cancelled, in entity.config().get().
+    // But it used to pass (e.g. with commit 56fcc1632ea4f5ac7f4136a7e04fabf501337540).
+    // It failed after the rename of CONF_MAP_THING_OBJ to CONF_MAP_OBJ_THING, which 
+    // suggests there was an underlying problem that was masked by the unfortunate naming
+    // of the previous "test.confMapThing.obj".
+    //
+    // Presumably an earlier call to task.get() timed out, causing it to cancel the task?
+    // I (Aled) question whether we want to support passing a task (rather than a 
+    // DeferredSupplier or TaskFactory, for example). Our EntitySpec.configure is overloaded
+    // to take a Task, but that feels wrong!?
```

Comment:
Passing Tasks as config keys makes sense for me (same as passing a future). Creating a task to compute some value and passing it to an entity is a common thing in Brooklyn.